bika/lims/controlpanel/bika_analysisservices.py

```diff
@@ -165,82 +183,72 @@ def __init__(self, context, request):
+        self.can_sort = not self.do_cats
+        self.currency_symbol = self.get_currency_symbol()
+        self.decimal_mark = self.get_decimal_mark()
         if self.do_cats:
             self.pagesize = 999999  # hide batching controls
             self.show_categories = True
             self.expand_all_categories = False
             self.ajax_categories = True
             self.category_index = "getCategoryTitle"
 
-        self.columns = {
-            "Title": {
+        self.columns = collections.OrderedDict((
+            ("Title", {
                 "title": _("Service"),
                 "index": "sortable_title",
                 "replace_url": "absolute_url",
-                "sortable": not self.do_cats,
-            },
-            "Keyword": {
+                "sortable": self.can_sort}),
+            ("Keyword", {
                 "title": _("Keyword"),
                 "index": "getKeyword",
                 "attr": "getKeyword",
-                "sortable": not self.do_cats,
-            },
-            "Category": {
+                "sortable": self.can_sort}),
+            ("Category", {
                 "title": _("Category"),
                 "attr": "getCategoryTitle",
-                "sortable": not self.do_cats,
-            },
-            "Methods": {
+                "sortable": self.can_sort}),
+            ("Methods", {
                 "title": _("Methods"),
-                "sortable": not self.do_cats,
-            },
-            "Department": {
+                "sortable": self.can_sort}),
+            ("Department", {
                 "title": _("Department"),
                 "toggle": False,
                 "attr": "getDepartment.Title",
-                "sortable": not self.do_cats,
-            },
-            "Unit": {
+                "sortable": self.can_sort}),
+            ("Unit", {
                 "title": _("Unit"),
                 "attr": "getUnit",
-                "sortable": False,
-            },
-            "Price": {
+                "sortable": False}),
+            ("Price", {
                 "title": _("Price"),
-                "sortable": not self.do_cats,
-            },
-            "MaxTimeAllowed": {
+                "sortable": self.can_sort}),
+            ("MaxTimeAllowed", {
                 "title": _("Max Time"),
                 "toggle": False,
-                "sortable": not self.do_cats,
-            },
-            "DuplicateVariation": {
+                "sortable": self.can_sort}),
+            ("DuplicateVariation", {
                 "title": _("Dup Var"),
                 "toggle": False,
-                "sortable": False,
-             },
-            "Calculation": {
+                "sortable": False}),
+            ("Calculation", {
                 "title": _("Calculation"),
-                "sortable": False,
-            },
-            "CommercialID": {
+                "sortable": False}),
+            ("CommercialID", {
```

Comment:
Column "Commercial ID" is not needed and can be removed from AnalysisService in future too